src/api/middleware/validators/company.js

```diff
@@ -55,18 +55,35 @@ const block = useExpressValidators([
         .trim(),
 ]);
 
-const unblock = useExpressValidators([
+const disable = useExpressValidators([
+    param("companyId")
+        .exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND),
+    body("adminReason")
+        .if((reason, { req }) => req.hasAdminPrivileges).exists().withMessage(ValidationReasons.REQUIRED).bail()
```

Comment:
I think the "has admin privileges" check should be done on the route level (not on the req validator). You can use an auth middleware on the route for this

Reply:
My line of thought was that the req validators were already checking for the existence of the "adminReason", which is not required from companies since they are not admins, and if the validator failed because we are not a privileged user there is no need to keep validating the "adminReason" . Mind if I ask why you think this should be put in a separate middleware instead?

Reply:
Wait... that's the thing: disabling a company does not require an adminReason, since companies can do that themselves (same logic as hiding an offer for example). Please remove this field from the validation entirely, as well as its usages on the CompanyService disable call.